pkg/azuredisk/azure_managedDiskController.go

```diff
@@ -414,6 +414,84 @@ func (c *ManagedDiskController) ResizeDisk(ctx context.Context, diskURI string,
 	return newSizeQuant, nil
 }
 
+// ModifyDisk: modify disk
+func (c *ManagedDiskController) ModifyDisk(ctx context.Context, options *ManagedDiskOptions) error {
+	klog.V(4).Infof("azureDisk - modifying managed Name:%s, StorageAccountType:%s, DiskIOPSReadWrite:%s, DiskMBpsReadWrite:%s", options.DiskName, options.StorageAccountType, options.DiskIOPSReadWrite, options.DiskMBpsReadWrite)
+
+	rg, subsID, err := getInfoFromDiskURI(options.SourceResourceID)
+	if err != nil {
+		return err
+	}
+
+	diskClient, err := c.clientFactory.GetDiskClientForSub(subsID)
+	if err != nil {
+		return err
+	}
+
+	model := armcompute.DiskUpdate{}
+	result, err := diskClient.Get(ctx, rg, options.DiskName)
+	if err != nil {
+		return err
+	}
+
+	if result.Properties == nil || result.SKU == nil || result.SKU.Name == nil {
+		return fmt.Errorf("DiskProperties or SKU of disk(%s) is nil", options.DiskName)
+	}
+
+	diskSku := *result.SKU.Name
+	if options.StorageAccountType != "" && options.StorageAccountType != diskSku {
```

Comment:
let's remove this check since if it's not allowed, CRP will return error finally, and this behavior may change in the future, we only need to doc somewhere.

Reply:
remove it, also add a doc for it